From 0a8e423b2d06a4604c51a1e5bfcc305203d49a62 Mon Sep 17 00:00:00 2001
Message-Id: <0a8e423b2d06a4604c51a1e5bfcc305203d49a62.1584650624.git.Jim.Somerville@windriver.com>
From: Jim Somerville <Jim.Somerville@windriver.com>
Date: Fri, 13 Mar 2020 16:15:29 -0400
Subject: [PATCH 1/2] Kernel source patches for TiC

Signed-off-by: Jim Somerville <Jim.Somerville@windriver.com>
Signed-off-by: Dongqi Chen <chen.dq@neusoft.com>
---
 SPECS/kernel-rt.spec | 30 ++++++++++++++++++++++++++++++
 1 file changed, 30 insertions(+)

diff --git a/SPECS/kernel-rt.spec b/SPECS/kernel-rt.spec
index c48b73e..c549d7e 100644
--- a/SPECS/kernel-rt.spec
+++ b/SPECS/kernel-rt.spec
@@ -432,6 +432,22 @@ Source3001: ima_signing_key.pub
 Source30000: kernel-rt-4.18.0-x86_64.config.tis_extra
 
 ## Patches needed for building this package
+# StarlingX Cloud patches here.
+Patch40002: Notification-of-death-of-arbitrary-processes.patch
+Patch40004: PCI-Add-ACS-quirk-for-Intel-Fortville-NICs.patch
+Patch40005: affine-compute-kernel-threads.patch
+Patch40006: Affine-irqs-and-workqueues-with-kthread_cpus.patch
+Patch40007: Make-kernel-start-eth-devices-at-offset.patch
+Patch40008: intel-iommu-allow-ignoring-Ethernet-device-RMRR-with.patch
+Patch40026: turn-off-write-same-in-smartqpi-driver.patch
+Patch40031: Allow-dmar-quirks-for-broken-bioses.patch
+
+# StarlingX Cloud rt patches here.
+
+Patch41000: debrand-rh-i686-cpu.patch
+Patch41001: debrand-rh_taint.patch
+Patch41002: debrand-single-cpu.patch
+Patch41003: restrict-iSCSI-kthreads-to-CPUs-in-cpu_kthread_mask.patch
 
 # END OF PATCH DEFINITIONS
 
@@ -937,6 +953,20 @@ cd linux-%{KVERREL}
 
 # END OF PATCH APPLICATIONS
 
+# StarlingX Cloud patches here.
+ApplyOptionalPatch Notification-of-death-of-arbitrary-processes.patch
+ApplyOptionalPatch PCI-Add-ACS-quirk-for-Intel-Fortville-NICs.patch
+ApplyOptionalPatch affine-compute-kernel-threads.patch
+ApplyOptionalPatch Affine-irqs-and-workqueues-with-kthread_cpus.patch
+ApplyOptionalPatch Make-kernel-start-eth-devices-at-offset.patch
+ApplyOptionalPatch intel-iommu-allow-ignoring-Ethernet-device-RMRR-with.patch
+ApplyOptionalPatch turn-off-write-same-in-smartqpi-driver.patch
+ApplyOptionalPatch Allow-dmar-quirks-for-broken-bioses.patch
+ApplyOptionalPatch debrand-rh-i686-cpu.patch
+ApplyOptionalPatch debrand-rh_taint.patch
+ApplyOptionalPatch debrand-single-cpu.patch
+ApplyOptionalPatch restrict-iSCSI-kthreads-to-CPUs-in-cpu_kthread_mask.patch
+
 # Any further pre-build tree manipulations happen here.
 
 %if %{with_realtime}
-- 
2.7.4

