From bcf3e28debee7cef9e37f4c750025530d99f943c Mon Sep 17 00:00:00 2001
From: Peng Zhang <Peng.Zhang2@windriver.com>
Date: Thu, 27 Apr 2023 19:34:33 +0000
Subject: [PATCH] Upgrade 5.10 kernel to 5.10.177

Update debian/changelog.

Signed-off-by: Peng Zhang <Peng.Zhang2@windriver.com>
---
 debian/changelog | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/debian/changelog b/debian/changelog
index 98186df..f454f64 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,3 +1,10 @@
+linux-rt (5.10.177-1) unstable; urgency=medium
+
+  * New upstream update:
+    https://git.yoctoproject.org/linux-yocto/log/?h=v5.10%2Fstandard%2Fpreempt-rt%2Fbase&qt=range&q=f6146935c0b2cb2d342d8d188e7779c8244883a3 
+
+ -- Peng Zhang <Peng.Zhang2@windriver.com>  Wed, 26 Apr 2023 19:13:42 +0000
+
 linux-rt (5.10.162-1) unstable; urgency=medium
 
   * New upstream update:
-- 
2.30.2

