From 282d6563fe01567af0719617127e29ef47b5cd7b Mon Sep 17 00:00:00 2001
From: Peng Zhang <Peng.Zhang2@windriver.com>
Date: Mon, 28 Aug 2023 06:12:16 +0000
Subject: [PATCH] Upgrade 5.10 kernel to 5.10.189

Update debian/changelog.

Signed-off-by: Peng Zhang <Peng.Zhang2@windriver.com>
---
 debian/changelog | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/debian/changelog b/debian/changelog
index c9a526a23..f31fd4012 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,3 +1,10 @@
+linux (5.10.189-1) unstable; urgency=medium
+
+  * New upstream update:
+    https://git.yoctoproject.org/linux-yocto/log/?h=v5.10%2Fstandard%2Fbase&qt=range&q=fc99f5e7ce23e4814c7b41f24c5d0296754be89e
+
+ -- Peng Zhang <Peng.Zhang2@windriver.com>  Mon, 28 Aug 2023 05:56:12 +0000
+
 linux (5.10.185-1) unstable; urgency=medium
 
   * New upstream update:
-- 
2.30.2

