From 2043610ac8bcfab43190e71d343779a44986857f Mon Sep 17 00:00:00 2001
From: Yong Fu <fuyong@neusoft.com>
Date: Wed, 27 May 2020 09:42:26 +0800
Subject: [PATCH] fix mlnx-ofa_kernel-devel dependency on python2

Signed-off-by: Yong Fu <fuyong@neusoft.com>
---
 SPECS/mlnx-ofa_kernel.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/mlnx-ofa_kernel.spec b/SPECS/mlnx-ofa_kernel.spec
index 2fccf4f..7525baa 100644
--- a/SPECS/mlnx-ofa_kernel.spec
+++ b/SPECS/mlnx-ofa_kernel.spec
@@ -112,6 +112,7 @@ Patch01: 0001-neuter-HAVE_IEEE_GETQCN.patch
 Patch02: 0001-neuter-a-bunch-of-ks-stuff.patch
 Patch03: 0001-neuter-some-stats-output.patch
 Patch04: Disable-use-of-page-pool.patch
+Patch05: Use-python3-instead-of-python2.patch
 
 BuildRoot: %{?build_root:%{build_root}}%{!?build_root:/var/tmp/OFED}
 Vendor: Mellanox Technologies
@@ -305,6 +306,7 @@ mkdir obj
 %patch02 -p1
 %patch03 -p1
 %patch04 -p1
+%patch05 -p1
 
 %build
 export EXTRA_CFLAGS='-DVERSION=\"%version\"'
-- 
2.17.1

