From 47ce98b70fcbf36998a0eb5995d13b73b1190708 Mon Sep 17 00:00:00 2001
From: Jiping Ma <jiping.ma2@windriver.com>
Date: Thu, 25 Aug 2022 22:10:18 -0700
Subject: [PATCH] Add BuildRequires on kernel-devel-keys

Signed-off-by: Jiping Ma <jiping.ma2@windriver.com>
---
 mlnx-ofa_kernel.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mlnx-ofa_kernel.spec b/mlnx-ofa_kernel.spec
index 2a1b956..9d869d5 100644
--- a/SPECS/mlnx-ofa_kernel.spec
+++ b/SPECS/mlnx-ofa_kernel.spec
@@ -33,7 +33,7 @@
 
 # KMP is disabled by default
 %{!?KMP: %global KMP 0}
-BuildRequires: kernel%{?bt_ext}-devel, openssl
+BuildRequires: kernel%{?bt_ext}-devel, kernel%{?bt_ext}-devel-keys, openssl
 BuildRequires:  devtoolset-8-build
 BuildRequires:  devtoolset-8-binutils
 BuildRequires:  devtoolset-8-gcc
-- 
2.35.1

