From 6c8e7bc564ccf4d2b4160f65b725a1550006ccd6 Mon Sep 17 00:00:00 2001
From: Jiping Ma <jiping.ma2@windriver.com>
Date: Sat, 29 Jan 2022 01:04:14 -0800
Subject: [PATCH] Support STX system

Signed-off-by: Jiping Ma <jiping.ma2@windriver.com>
---
 mlnx-tools.spec | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/SPECS/mlnx-tools.spec b/SPECS//mlnx-tools.spec
index 196721f..6535863 100644
--- a/SPECS/mlnx-tools.spec
+++ b/SPECS/mlnx-tools.spec
@@ -25,11 +25,16 @@
 # and/or other materials provided with the distribution.
 #
 #
+%if "%{_tis_build_type}" == "rt"
+%define bt_ext -rt
+%else
+%undefine bt_ext
+%endif
 
 Summary: Mellanox userland tools and scripts
 Name: mlnx-tools
 Version: 5.2.0
-Release: 0%{?_dist}.55103
+Release: 0.55103%{?_tis_dist}.%{tis_patch_ver}
 License: GPLv2
 Url: https://github.com/Mellanox/mlnx-tools
 Group: Applications/System
-- 
2.31.1

