From 3be2d62d2671002aaf14eae0a9759ee99f1d8eae Mon Sep 17 00:00:00 2001
From: Jiping Ma <jiping.ma2@windriver.com>
Date: Sat, 29 Jan 2022 01:49:48 -0800
Subject: [PATCH] Support STX system

Signed-off-by: Jiping Ma <jiping.ma2@windriver.com>
---
 rdma-core.spec | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/SPECS/rdma-core.spec b/SPECS/rdma-core.spec
index 7f1a9ee..5b1e337 100644
--- a/SPECS/rdma-core.spec
+++ b/SPECS/rdma-core.spec
@@ -1,3 +1,9 @@
+%if "%{_tis_build_type}" == "rt"
+%define bt_ext -rt
+%else
+%undefine bt_ext
+%endif
+
 %{!?cmake: %global cmake cmake}
 %{!?make_jobs: %global make_jobs make VERBOSE=1 %{?_smp_mflags}}
 %{!?cmake_install: %global cmake_install DESTDIR=%{buildroot} make install}
@@ -27,7 +33,7 @@
 
 Name: rdma-core
 Version: 55mlnx37
-Release: 1%{?dist}.55103
+Release: 1.55103%{?_tis_dist}.%{tis_patch_ver}
 Summary: RDMA core userspace libraries and daemons
 Group: System Environment/Libraries
 
@@ -97,6 +103,7 @@ BuildRequires: python-docutils
 BuildRequires: perl-generators
 %endif
 
+BuildRequires: cmake
 # Red Hat/Fedora previously shipped redhat/ as a stand-alone
 # package called 'rdma', which we're supplanting here.
 Provides: rdma = %{version}-%{release}
-- 
2.31.1

