From 5c639bdbb7e4a652537d743b6303ce2c20eafe2e Mon Sep 17 00:00:00 2001
From: Jiping Ma <jiping.ma2@windriver.com>
Date: Thu, 25 Aug 2022 23:56:58 -0700
Subject: [PATCH] Add BuildRequires on kernel-devel-keys

Signed-off-by: Jiping Ma <jiping.ma2@windriver.com>
---
 opae-intel-fpga-driver.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/opae-intel-fpga-driver.spec b/opae-intel-fpga-driver.spec
index 193f149..d5d73b7 100755
--- a/SPECS/opae-intel-fpga-driver.spec
+++ b/SPECS/opae-intel-fpga-driver.spec
@@ -23,7 +23,7 @@ Source: %{kmod_name}-%{version}.tar.gz
 Exclusiveos: linux
 ExclusiveArch: i386 i586 i686 x86_64
 Buildroot: %{_builddir}/%{kmod_name}-%{version}
-BuildRequires: kernel%{?bt_ext}-devel, tar, gcc, make
+BuildRequires: kernel%{?bt_ext}-devel, kernel%{?bt_ext}-devel-keys, tar, gcc, make
 %if 0%{?rhel} == 7
 BuildRequires:  devtoolset-8-build
 BuildRequires:  devtoolset-8-binutils
-- 
2.35.1

