From ac67546ccbecd3ee18145be31d87d7253048adda Mon Sep 17 00:00:00 2001
Message-Id: <ac67546ccbecd3ee18145be31d87d7253048adda.1566587341.git.Jim.Somerville@windriver.com>
From: "zhao.shuai" <zhaos@neusoft.com>
Date: Mon, 5 Aug 2019 17:55:01 +0800
Subject: [PATCH 2/3] Kernel-source-patches-for-TiC

Signed-off-by: zhao.shuai <zhaos@neusoft.com>
Signed-off-by: Jim Somerville <Jim.Somerville@windriver.com>
Signed-off-by: Robin Lu <bin1.lu@intel.com>
Signed-off-by: Zhang Zhiguo <zhangzhg@neusoft.com>
Signed-off-by: Shuicheng Lin <shuicheng.lin@intel.com>
Signed-off-by: Dongqi Chen <chen.dq@neusoft.com>
---
 SPECS/kernel.spec | 22 ++++++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/SPECS/kernel.spec b/SPECS/kernel.spec
index adc3580..a7a8c97 100644
--- a/SPECS/kernel.spec
+++ b/SPECS/kernel.spec
@@ -478,6 +478,18 @@ Patch1001: debrand-rh_taint.patch
 # empty final patch to facilitate testing of kernel patches
 Patch999999: linux-kernel-test.patch
 
+# StarlingX Cloud patches here.
+Patch40001: Notification-of-death-of-arbitrary-processes.patch
+Patch40002: PCI-Add-ACS-quirk-for-Intel-Fortville-NICs.patch
+Patch40003: affine-compute-kernel-threads.patch
+Patch40004: Affine-irqs-and-workqueues-with-kthread_cpus.patch
+Patch40005: Make-kernel-start-eth-devices-at-offset.patch
+Patch40006: intel-iommu-allow-ignoring-Ethernet-device-RMRR-with.patch
+# DRBD was choking on write same
+Patch40009: turn-off-write-same-in-smartqpi-driver.patch
+# Workaround for broken bios causing IOMMU issues
+Patch40010: Allow-dmar-quirks-for-broken-bioses.patch
+
 # END OF PATCH DEFINITIONS
 
 BuildRoot: %{_tmppath}/kernel-%{KVERREL}-root
@@ -1000,6 +1012,16 @@ ApplyOptionalPatch debrand-rh_taint.patch
 #ApplyOptionalPatch debrand-rh-i686-cpu.patch 
 
 
+# StarlingX Cloud patches here.
+ApplyOptionalPatch Notification-of-death-of-arbitrary-processes.patch
+ApplyOptionalPatch PCI-Add-ACS-quirk-for-Intel-Fortville-NICs.patch
+ApplyOptionalPatch affine-compute-kernel-threads.patch
+ApplyOptionalPatch Affine-irqs-and-workqueues-with-kthread_cpus.patch
+ApplyOptionalPatch Make-kernel-start-eth-devices-at-offset.patch
+ApplyOptionalPatch intel-iommu-allow-ignoring-Ethernet-device-RMRR-with.patch
+ApplyOptionalPatch turn-off-write-same-in-smartqpi-driver.patch
+ApplyOptionalPatch Allow-dmar-quirks-for-broken-bioses.patch
+
 # END OF PATCH APPLICATIONS
 
 # Any further pre-build tree manipulations happen here.
-- 
2.7.4

